tests/h/views/api/groups_test.py

```diff
@@ -184,6 +186,78 @@ def pyramid_request(self, pyramid_request, factories):
         return pyramid_request
 
 
+@pytest.mark.usefixtures('group_service',
+                         'user_service',
+                         'request_auth_client',
+                         'validate_auth_client_authority')
+class TestAddMember(object):
+
+    def test_it_adds_current_user_to_group(self,
+                                           factories,
+                                           pyramid_request,
+                                           group_service,
+                                           user_service):
+        group = factories.Group()
+        user = factories.User()
+
+        user_service.fetch.return_value = user
+
+        pyramid_request.matchdict['userid'] = user.userid
+        pyramid_request.matchdict['pubid'] = group.pubid
+
+        resp = views.add_member(group, pyramid_request)
+
+        group_service.member_join.assert_called_once_with(group, user.userid)
+        assert isinstance(resp, HTTPNoContent)
+
+    def test_it_fails_with_mismatched_user_and_group_authorities(self,
+                                                                 factories,
+                                                                 pyramid_request,
+                                                                 group_service,
+                                                                 user_service):
+        group = factories.Group(authority="different_authority.com")
+        user = factories.User()
+
+        user_service.fetch.return_value = user
+
+        pyramid_request.matchdict['userid'] = user.userid
+        pyramid_request.matchdict['pubid'] = group.pubid
+
+        with pytest.raises(HTTPNotFound):
+            views.add_member(group, pyramid_request)
+
+    def test_it_fails_with_non_existent_user(self,
+                                             factories,
+                                             pyramid_request,
+                                             group_service,
+                                             user_service):
+        group = factories.Group()
+
+        user_service.fetch.return_value = None
+
+        pyramid_request.matchdict['userid'] = "some_user"
+        pyramid_request.matchdict['pubid'] = group.pubid
+
+        with pytest.raises(HTTPNotFound):
+            views.add_member(group, pyramid_request)
+
+    def test_it_fails_with_non_existent_group(self,
```

Comment:
I am curious to know whether the 404 is raised at line 100 of the view or by the traversal failure/group lookup fail before the view execution happens...